### DISCUSSION TITLE
Symbol and name issues in bridging RING to zkSync and Scroll?

### DISCUSSION THREAD

**Discussion by hackfisher** [ringdao-community-guild/discussion#89] [Integration]
https://explorer.zksync.io/address/0xc99A8De2aCFEfFcE1eB595EAbD28b449924f1F95#contract

zkSync and Scroll may only support L2 token with symbol and names being type of string instead of byte32?

No better solution, one possibility is that next Helix xToken version will using a new xRING token contract (also consideration of better standard and gas consume, and new xRING will use string as type of symbol and name, new name can simply use "Darwinia Token")

Current RING contract can continue work by working as a wrap token of xRING token, a Wrap contract(no owner) will be deployed to manage this, with issue/destroy auth to RING contract. In this way, current interface to exchanges and data platform will not need emergency/hurry handling, the RING contract can still be widely supported and used.

@hujw77 @xiaoch05 Could you help review this?

**Comment by hackfisher** [comment]
RING contract's name may need to changed and be update to reflect that it is  Wrap token of xRING e.g. "Wrap token of Darwinia xRING"

And xRING token better to support unlimited approve to Wrap token to save the gas and better experience of wrap()/wrap_to(_to).

**Comment by hackfisher** [comment]
related: https://github.com/helix-bridge/contracts/issues/55

**Comment by Bertachi** [comment]
all right full losse
